src/cli/config_validation.go

```diff
@@ -0,0 +1,227 @@
+/*
+* ZDNS Copyright 2024 Regents of the University of Michigan
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not
+* use this file except in compliance with the License. You may obtain a copy
+* of the License at http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+* implied. See the License for the specific language governing
+* permissions and limitations under the License.
+ */
+package cli
+
+import (
+	"fmt"
+	"net"
+	"net/netip"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/zmap/dns"
+
+	"github.com/zmap/zdns/src/internal/util"
+	"github.com/zmap/zdns/src/zdns"
+)
+
+const (
+	loopbackAddrSubnet = "127.0.0.0/8"
```

Comment:
Do we need to handle IPv6 here too?

Reply:
I'll handle this when we handle adding IPv6 [support](https://github.com/zmap/zdns/issues/304). I'll add a `TODO` here tho so I don't forget to update this.